### PR TITLE
Add Provided APIs to PackageManifest

### PR DESCRIPTION
**Pull request by alecmerdler**
### Description

Allows clients (like the UI) to compare a `PackageManifest` against a list of `OperatorGroups` to determine if the APIs are already provided by an existing Operator before creating a `Subscription.

### Screenshots

**JSON response:**
Screenshot_20190315_170939

Addresses https://jira.coreos.com/browse/OLM-954

### CODE REVIEW

**Comment by alecmerdler** [comment]
I ran `make container-codegen`, but looks like it didn't work. From the `packageserver` pod logs:
```
Error: unable to setup API &{[packages.apps.redhat.com/v1alpha1] map[v1alpha1:map[packagemanifests:0xc4207a43c0]] v1 <nil> 0xc4200b7e00 {0xc4200b7e00 [{[application/json] application/json [json] true 0xc42039b780 0xc42039b7c0 []  {} 0xc42039b780} {[application/yaml] application/yaml [yaml] true 0xc42039b800 <nil> []  <nil> <nil>} {[application/vnd.kubernetes.protobuf] application/vnd.kubernetes.protobuf [pb] false 0xc4202d9860 <nil> []  {} 0xc420213ec0}] 0xc420346d60 [{application/json true 0xc42039b780 0xc42039b7c0 0xc420213f20} {application/yaml true 0xc42039b800 <nil> <nil>} {application/vnd.kubernetes.protobuf false 0xc4202d9860 <nil> 0xc420213f50}] 0xc42039b780} 0xc420446f40}: error in registering resource: packagemanifests, unable to build openapi definitions for packages.apps.redhat.com/v1alpha1, Kind=PackageManifest: cannot find model definition for github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1.APIServiceDefinitions. If you added a new type, you may need to add +k8s:openapi-gen=true to the package or type and run code-gen again
```

File: Makefile [hunk 1]
```diff
@@ -122,7 +122,7 @@ codegen:
 	cp scripts/generate_groups.sh vendor/k8s.io/code-generator/generate_groups.sh
 	mkdir -p vendor/k8s.io/code-generator/hack
 	cp boilerplate.go.txt vendor/k8s.io/code-generator/hack/boilerplate.go.txt
-	go run vendor/k8s.io/kube-openapi/cmd/openapi-gen/openapi-gen.go --logtostderr -i ./vendor/k8s.io/apimachinery/pkg/runtime,./vendor/k8s.io/apimachinery/pkg/apis/meta/v1,./vendor/k8s.io/apimachinery/pkg/version,./pkg/package-server/apis/packagemanifest/v1alpha1,./pkg/api/apis/operators/v1alpha1 -p $(PKG)/pkg/package-server/apis/openapi -O zz_generated.openapi -h boilerplate.go.txt -r /dev/null
+	go run vendor/k8s.io/kube-openapi/cmd/openapi-gen/openapi-gen.go --logtostderr -i ./vendor/k8s.io/apimachinery/pkg/runtime,./vendor/k8s.io/apimachinery/pkg/apis/meta/v1,./vendor/k8s.io/apimachinery/pkg/version,./pkg/api/apis/operators/v1alpha1,./pkg/package-server/apis/packagemanifest/v1alpha1 -p $(PKG)/pkg/package-server/apis/openapi -O zz_generated.openapi -h boilerplate.go.txt -r /dev/null
```
Comment (alecmerdler) [inline Makefile]: Had to change ordering in order to use `ClusterServiceVersion` types in `PackageManifest` model.

File: test/e2e/installplan_e2e_test.go [hunk 2]
```diff
@@ -172,8 +172,8 @@ func newCSV(name, namespace, replaces string, version semver.Version, owned []ap
 			},
 			InstallStrategy: namedStrategy,
 			CustomResourceDefinitions: v1alpha1.CustomResourceDefinitions{
-				Owned:    []v1alpha1.CRDDescription{},
-				Required: []v1alpha1.CRDDescription{},
+				Owned:    nil,
```
Comment (alecmerdler) [inline test/e2e/installplan_e2e_test.go]: Generated clients seem to deserialize `omitempty` into `nil` slices, which breaks comparisons with empty slices. The `append()` function works with `nil` slices anyway, so this _shouldn't_ negatively affect anything.

**Comment by ecordell** [comment]
@alecmerdler this needs a rebase / regen

We added support for serializing/deserializing the version field, which changed a bunch of the codegen.

**Comment by ecordell** [comment]
/hold

holding until master is open for 4.2 😢

**Comment by alecmerdler** [comment]
The `make unit` tests pass fine for me locally, not sure what the issue is on CI.

**Pull request merged**